Doc/library/token.rst

```diff
@@ -44,40 +48,157 @@ functions.  The functions mirror definitions in the Python C header files.
+   Note that keywords are also initially tokenized an ``NAME`` tokens.
+
+.. data:: NUMBER
+
+   Token value that indicates a :ref:`numeric literal <numbers>`
+
+.. data:: STRING
+
+   Token value that indicates a :ref:`string or byte literal <strings>`,
+   excluding :ref:`formatted string literals <f-strings>`.
+   The token string is not interpreted:
+   it includes the surrounding quotation marks and the prefix (if given);
+   backslashes are included literally, without processing escape sequences.
 
-The following token type values aren't used by the C tokenizer but are needed for
-the :mod:`tokenize` module.
+.. data:: OP
+
+   A generic token value that indicates an
+   :ref:`operator <operators>` or :ref:`delimiter <delimiters>`.
+
+   .. impl-detail::
+
+      This value is only reported by the :mod:`tokenize` module.
+      Internally, the tokenizer uses
+      :ref:`exact token types <token_operators_delimiters>` instead.
 
 .. data:: COMMENT
-   :noindex:
 
    Token value used to indicate a comment.
+   The parser ignores :data:`!COMMENT` tokens.
 
+.. data:: NEWLINE
+
+   Token value that indicates the end of a :ref:`logical line <logical-lines>`.
 
 .. data:: NL
-   :noindex:
 
-   Token value used to indicate a non-terminating newline.  The
-   :data:`NEWLINE` token indicates the end of a logical line of Python code;
-   ``NL`` tokens are generated when a logical line of code is continued over
-   multiple physical lines.
+   Token value used to indicate a non-terminating newline.
+   :data:`!NL` tokens are generated when a logical line of code is continued
+   over multiple physical lines. The parser ignores :data:`!NL` tokens.
+
+.. data:: INDENT
+
+   Token value used at the beginning of a :ref:`logical line <logical-lines>`
+   to indicate the start of an :ref:`indented block <indentation>`.
+
+.. data:: DEDENT
+
+   Token value used at the beginning of a :ref:`logical line <logical-lines>`
+   to indicate the end of an :ref:`indented block <indentation>`.
+
+.. data:: FSTRING_START
+
+   Token value used to indicate the beginning of an
+   :ref:`f-string literal <f-strings>`.
+
+   .. impl-detail::
+
+      The token string includes the prefix and the opening quote(s), but none
+      of the contents of the literal.
+
+.. data:: FSTRING_MIDDLE
+
+   Token value used for literal text inside an :ref:`f-string literal <f-strings>`,
+   including format specifications.
+
+   .. impl-detail::
+
+      Replacement fields (that is, the non-literal parts of f-strings) use
+      the same tokens as other expressions, and are delimited by
+      :data:`LBRACE`, :data:`RBRACE`, :data:`EXCLAMATION` and :data:`COLON`
+      tokens.
+
+.. data:: FSTRING_END
+
+   Token value used to indicate the end of a :ref:`f-string <f-strings>`.
 
+   .. impl-detail::
+
+      The token string contains the closing quote(s).
+
+.. data:: ENDMARKER
+
+   Token value that indicates the end of input.
+   Used in :ref:`top-level grammar rules <top-level>`.
```

Comment:
The 3.13 docs use a simplified version of grammar, which does not use ENDMARKER.
```suggestion
```